src/react/useColorPalettes.ts

```diff
@@ -0,0 +1,46 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useEffect, useState } from 'react';
+import { AppBridgeBlock } from '../AppBridgeBlock';
+import { FrontifyColorPalette } from '../types/FrontifyColorPalette';
+
+export type UseColorPalettesReturnType = {
+    colorPalettes: FrontifyColorPalette[];
+};
+
+export const useColorPalettes = (appBridge: AppBridgeBlock, colorPaletteIds?: number[]): UseColorPalettesReturnType => {
+    const [colorPalettes, setColorPalettes] = useState<FrontifyColorPalette[]>([]);
+
+    useEffect(() => {
+        const getColorPalettes = async () => {
+            try {
+                const colorPalettes: FrontifyColorPalette[] = await appBridge.getColorPalettes();
+
+                if (colorPaletteIds) {
+                    const filteredColorPalettes = colorPalettes.filter((colorPalette: FrontifyColorPalette) =>
+                        colorPaletteIds.includes(colorPalette.id),
+                    );
+
+                    await Promise.all(
+                        filteredColorPalettes.map(
+                            async (colorPalette: FrontifyColorPalette) =>
```

Comment:
```suggestion
                            async (colorPalette) =>
```
Type should be implicitely set already